ekf/tests/sensortest.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Phoenix-Pilot
+ *
+ * extended kalman filter
+ * 
+ * ekf sensor reading test that aims to measure noise levels in sensor readings
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-Pilot software
+ *
+ * %LICENSE%
+ */
+
+#include <board_config.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <sys/threads.h>
+
+#include "../sensc.h"
+#include <ekflib.h>
+
+#define R2D           57.2957
+#define THROTTLE_MAX  50
+#define THROTTLE_MIN  10
+#define NUM_OF_MOTORS 4
+
+struct {
+	int m[NUM_OF_MOTORS];
+	volatile int run;
+	char stack[2048];
+	char throttle[7];
+} sensortest_common;
+
+
+static void sensortest_helpPrint(const char *appName)
+{
+	printf("Usage: %s [throttle]\n", appName);
+	printf("  throttle - integer in range [%d,%d], interpreted as percents\n", THROTTLE_MIN, THROTTLE_MAX);
+}
+
+
+static int sensortest_allWrite(const char *val, int sz)
+{
+	int i;
+	for (i = 0; i < NUM_OF_MOTORS; i++) {
+		if (write(sensortest_common.m[i], val, sz) < 0) {
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+static void sensortest_allClose(void)
+{
+	int i;
+
+	for (i = 0; i < NUM_OF_MOTORS; i++) {
+		if (sensortest_common.m[i] < 0) {
+			close(sensortest_common.m[i]);
+		}
+	}
+}
+
+static void sensortest_thread(void *arg)
+{
+	sensor_event_t aEvt, gEvt, mEvt;
+	ekf_state_t uavState;
+
+	if (ekf_init() == 0) {
+		ekf_run();
+	}
+
+
+	while (sensortest_common.run > 0) {
+		usleep(1000 * 10);
+		sensc_imuGet(&aEvt, &gEvt, &mEvt);
+		ekf_stateGet(&uavState);
+		printf("%f %f %f %d %d %d %d %d %d %f %f %f\n", uavState.accelX, uavState.accelY, uavState.accelZ, gEvt.gyro.gyroX, gEvt.gyro.gyroY, gEvt.gyro.gyroZ, mEvt.mag.magX, mEvt.mag.magY, mEvt.mag.magZ, uavState.yaw * R2D, uavState.pitch * R2D, uavState.roll * R2D);
+	}
+
+	endthread();
+}
+
+
+static void sensortest_emergencyShutdown(const char *shutdownMsg, int msgSz)
+{
+	unsigned int motorShdwn[NUM_OF_MOTORS] = { 0 };
+	int i = 0, flag = 0;
+
+	while (flag == 0) {
+		flag = 1;
+		for (i = 0; i < NUM_OF_MOTORS; i++) {
+			if ((motorShdwn[i] == 0) && (write(sensortest_common.m[i], shutdownMsg, msgSz) < 0)) {
+				flag = 0;
+				printf("Emergency shutdown failed, retry...\n");
+			}
+			else {
+				motorShdwn[i] = 1;
+			}
+		}
+	}
+}
+
+
+static int sensortest_motorsProcedure(void)
+{
+	const char pwmInit[] = "100000";
+
+	if (sensortest_allWrite(pwmInit, sizeof(pwmInit)) < 0) {
+		sensortest_emergencyShutdown(pwmInit, sizeof(pwmInit));
+		return -1;
+	}
+	sleep(9);
+	if (sensortest_allWrite(sensortest_common.throttle, sizeof(sensortest_common.throttle)) < 0) {
+		sensortest_emergencyShutdown(pwmInit, sizeof(pwmInit));
+		return -1;
+	}
+	sleep(7);
+	if (sensortest_allWrite(pwmInit, sizeof(pwmInit)) < 0) {
+		sensortest_emergencyShutdown(pwmInit, sizeof(pwmInit));
+		return -1;
+	}
+
+	return 0;
+}
+
+
+int main(int argc, char **argv)
+{
+	unsigned int power, i, flag = 0, ret;
```

Comment:
```suggestion
	unsigned int throttle, i, flag = 0, ret;
```
To be compliant with our convention.